selftests/functional/test_basic.py

```diff
@@ -235,31 +235,6 @@ def test_runner_ignore_missing_references_all_missing(self):
         self.assertEqual(result.exit_status, expected_rc,
                          "Avocado did not return rc %d:\n%s" % (expected_rc, result))
 
-    @unittest.skipIf(not CC_BINARY,
```

Comment:
I'm wondering when this feature was removed? As far as I know it was just altered in 327ad2142fb9acae2eca11379a672ce8b022f530 to not to include `.py` automatically, but apart from that it still works. Prove is that the selftests are passing, I'm actually using `avocado run passtest.py` and there is the code in `avocado.core.loader.FileLoader._make_tests` which uses the `data_dir.get_test_dir()`. I think this selftest should be kept or ideally replaced with unittest to discover only (why to run tests just to see they are discoverable, especially when we run the same tests on different places using the same style)

Reply:
I think you're missing the fact that this test was passing because the test reference was changed to include `.py`.  Thus, effectively undermining the original goal of the test.  I'm not sure what you mean about the datadir... this test was only using `examples/tests/datadir.py`.

Reply:
I think I see the confusion. There are in-deed duplicate tests `test_datadir_noalias` and `test_runner_noalias` which serve (almost) the same purpose, but in general those tests are the only place where test test-aliases which are still supported in slightly modified way. The test-alias used to turn `passtest` into `$test_dir/passtest.py`. We found this incorrect as Avocado should treat all tests types the same way so test-alias was turned into prefixing `$test_dir` to the passed id, therefor `test_dir` => `$test_dir/passtest`, which should allow running any file from test_dir, which is why the 2 `.sh` selftests were added.

For a temporary period of time we also allowed `passtest` to translate to `$test_dir/passtest.py` if `$test_dir/passtest` did not exist but looking at those tests we either removed it (or haven't had implemented).

So I think we should revert this and only remove or skip the `test_runner_noalias(self)` in case `CC_BINARY` not available. (basically I'm trying to say we should keep at least `alias/noalias` execution of `.py/.sh` files, therefor 4 tests. Yes, IMO we could do it by adding comment in some other tests to avoid executing the same things mulitple times, but this was rejected previously so we should keep at least those 4 tests.